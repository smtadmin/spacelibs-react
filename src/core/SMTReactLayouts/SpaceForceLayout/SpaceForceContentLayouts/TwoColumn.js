/*
 * File: TwoColumn
 * Version: 0.1.0
 * Project: react-test
 * Description: Two column layout for Spaceforce content. It is defauled to a col-2 and col-10 that can be overriden with whatever bootstrap
 * you would like to specify when rendering the TwoColumn Component in your page
 * File Created: Wednesday, 3rd February 2021 3:56 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 10th February 2021 4:16 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import React from 'react';
import BootstrapWrapper from '../BootstrapWrapper/BootstrapWrapper';
import { Container, Row, Col } from 'react-bootstrap';

/**
 * Two column layout for Spaceforce content.
 */
class TwoColumn extends React.Component {

    /**
     * Creates an instance of the TwoColumn class
     * @param {*} props 
     */
    constructor(props) {
        super(props);
        this.colOneBootstrap = (typeof props.colOneBootstrap === 'undefined') ? "col-2" : props.colOneBootstrap;
        this.colTwoBootstrap = (typeof props.colTwoBootstrap === 'undefined') ? "col" : props.colTwoBootstrap;
        this.state = {
            colOne : this.addKeys("colOne", this.props.colOne),
            colTwo : this.addKeys("colTwo", this.props.colTwo)
        }
    }

    /**
     * Adds unique keys to each of the content elements passted to it
     * @param {*} col Column Label
     * @param {*} columnContent Elements to be placed in the columns
     */
    addKeys(col, columnContent) {
        const newElements = [];
        let count = 0;

        if (Array.isArray(columnContent)) {
            for (const element of columnContent) {
                newElements.push(<div key={count}>{element}</div>);
                count++;
            }
        } else {
            newElements.push(<div key={count}>{columnContent}</div>);
        }

        return <BootstrapWrapper sectionName={col} content={newElements} />;
    }
  
    /**
     * Returns a TwoColumn Component
     */
    render() {
        return (
            <Container fluid className="two-column-container">
                <Row>
                    <Col key="col-1" className={"col-one " + this.colOneBootstrap}>
                        {this.state.colOne}
                    </Col>
                    <Col key="col-2" className={"col-two " + this.colTwoBootstrap}>
                        {this.state.colTwo}
                    </Col>
                </Row>
            </Container>
        )
    }

}

export default TwoColumn;