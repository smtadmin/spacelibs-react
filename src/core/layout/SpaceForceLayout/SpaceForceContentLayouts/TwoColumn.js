/*
 * File: TwoColumn
 * Version: 0.1.0
 * Project: react-test
 * Description: Two column layout for Spaceforce content. It is defauled to a col-2 and col-10 that can be overriden with whatever bootstrap
 * you would like to specify when rendering the TwoColumn Component in your page
 * File Created: Wednesday, 3rd February 2021 3:56 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Thursday, 18th March 2021 11:40 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import React from 'react';
import PropTypes from 'prop-types';
import BootstrapWrapper from '../BootstrapWrapper/BootstrapWrapper';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

/**
 * Two column layout for Spaceforce content.
 */
class TwoColumn extends React.Component {
    /**
     * Creates an instance of the TwoColumn class
     * @param {*} props TwoColumn Props
     */
    constructor(props) {
        super(props);
        this.colOneBootstrap =
            typeof props.colOneBootstrap === "undefined"
                ? "col-2"
                : props.colOneBootstrap;
        this.colTwoBootstrap =
            typeof props.colTwoBootstrap === "undefined"
                ? "col"
                : props.colTwoBootstrap;
        this.state = {
            colOne: this.addKeys("colOne", this.props.colOne),
            colTwo: this.addKeys("colTwo", this.props.colTwo),
        };
    }

    /**
     * Adds unique keys to each of the content elements passted to it
     *
     * @param {*} col Column Label
     * @param {*} columnContent Elements to be placed in the columns
     * @returns {*} Bootstrap wrapper with keys set
     */
    addKeys(col, columnContent) {
        const newElements = [];
        let count = 0;

        if (Array.isArray(columnContent)) {
            for (const element of columnContent) {
                newElements.push(<div key={count}>{element}</div>);
                count++;
            }
        } else {
            newElements.push(<div key={count}>{columnContent}</div>);
        }

        return <BootstrapWrapper sectionName={col} content={newElements} />;
    }

    /**
     * Returns a TwoColumn Component
     *
     * @returns {*} React components
     * @memberof TwoColumn
     */
    render() {
        return (
            <Container fluid className='two-column-container'>
                <Row>
                    <Col
                        key='col-1'
                        className={"col-one " + this.colOneBootstrap}>
                        {this.state.colOne}
                    </Col>
                    <Col
                        key='col-2'
                        className={"col-two " + this.colTwoBootstrap}>
                        {this.state.colTwo}
                    </Col>
                </Row>
            </Container>
        );
    }
}

/**
 * Accepted Prop Types
 */
TwoColumn.propTypes = {
    colOne: PropTypes.any,
    colTwo: PropTypes.any,
    colOneBootstrap: PropTypes.any,
    colTwoBootstrap: PropTypes.any
};

export default TwoColumn;