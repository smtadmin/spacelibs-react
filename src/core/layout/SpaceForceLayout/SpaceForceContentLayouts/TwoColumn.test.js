/*
 * File: TwoColumn.test
 * Version: 0.1.0
 * Project: react-test
 * Description: Testing suite for the Space Force Content Layouts Two Column Layout
 * File Created: Monday, 8th February 2021 12:34 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Friday, 12th February 2021 9:15 am
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import TwoColumn from './TwoColumn';
import React from 'react';
import { render } from '@testing-library/react';

/**
 * It renders the TwoColumn element without crashing
 */
it("Renders without crashing", () => {

    const element = render(<TwoColumn  />);
    expect(element).toBeTruthy();

});

/**
 * It renders a single element into the TwoColumn layout correctly
 */
it("Renders The Layout Correctly With One Element In Each Columnm", () => {

    const element = render(<TwoColumn colOne={<div>Hello World</div>} colTwo={<div>Goodbye World</div>} />);
    expect(element).toMatchSnapshot();
})

/**
 * It renders multiple elements into the TwoColumn layout correctly
 */
it("Renders The Layout Correctly With Multiple Element", () => {

    const element = render(<TwoColumn colOne={[<div>One</div>, <div>Two</div>, <div>Three</div>]} />);
    expect(element).toMatchSnapshot();
})

/**
 * It renders multiple elements into the TwoColumn layout correctly with bootstrap overrides
 */
it("Renders The Layout Correctly With Overridden Bootstrap", () => {

    const element = render(<TwoColumn colOneBootstrap="d-none" colTwoBootstrap="d-none" colOne={[<div>One</div>, <div>Two</div>, <div>Three</div>]} />);
    expect(element).toMatchSnapshot();
})