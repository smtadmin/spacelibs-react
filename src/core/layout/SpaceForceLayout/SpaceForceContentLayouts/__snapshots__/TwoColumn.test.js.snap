// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders The Layout Correctly With Multiple Element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="two-column-container container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-one col-2 col"
          >
            <div
              class="colOne-direction-wrapper  row"
            >
              <div
                class="colOne-header-element-0 col"
              >
                <div>
                  <div>
                    One
                  </div>
                </div>
              </div>
              <div
                class="colOne-header-element-1 col"
              >
                <div>
                  <div>
                    Two
                  </div>
                </div>
              </div>
              <div
                class="colOne-header-element-2 col"
              >
                <div>
                  <div>
                    Three
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-two col col"
          >
            <div
              class="colTwo-direction-wrapper  row"
            >
              <div
                class="colTwo-header-element-0 col"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="two-column-container container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-one col-2 col"
        >
          <div
            class="colOne-direction-wrapper  row"
          >
            <div
              class="colOne-header-element-0 col"
            >
              <div>
                <div>
                  One
                </div>
              </div>
            </div>
            <div
              class="colOne-header-element-1 col"
            >
              <div>
                <div>
                  Two
                </div>
              </div>
            </div>
            <div
              class="colOne-header-element-2 col"
            >
              <div>
                <div>
                  Three
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-two col col"
        >
          <div
            class="colTwo-direction-wrapper  row"
          >
            <div
              class="colTwo-header-element-0 col"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders The Layout Correctly With One Element In Each Columnm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="two-column-container container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-one col-2 col"
          >
            <div
              class="colOne-direction-wrapper  row"
            >
              <div
                class="colOne-header-element-0 col"
              >
                <div>
                  <div>
                    Hello World
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-two col col"
          >
            <div
              class="colTwo-direction-wrapper  row"
            >
              <div
                class="colTwo-header-element-0 col"
              >
                <div>
                  <div>
                    Goodbye World
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="two-column-container container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-one col-2 col"
        >
          <div
            class="colOne-direction-wrapper  row"
          >
            <div
              class="colOne-header-element-0 col"
            >
              <div>
                <div>
                  Hello World
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-two col col"
        >
          <div
            class="colTwo-direction-wrapper  row"
          >
            <div
              class="colTwo-header-element-0 col"
            >
              <div>
                <div>
                  Goodbye World
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders The Layout Correctly With Overridden Bootstrap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="two-column-container container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-one d-none col"
          >
            <div
              class="colOne-direction-wrapper  row"
            >
              <div
                class="colOne-header-element-0 col"
              >
                <div>
                  <div>
                    One
                  </div>
                </div>
              </div>
              <div
                class="colOne-header-element-1 col"
              >
                <div>
                  <div>
                    Two
                  </div>
                </div>
              </div>
              <div
                class="colOne-header-element-2 col"
              >
                <div>
                  <div>
                    Three
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-two d-none col"
          >
            <div
              class="colTwo-direction-wrapper  row"
            >
              <div
                class="colTwo-header-element-0 col"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="two-column-container container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-one d-none col"
        >
          <div
            class="colOne-direction-wrapper  row"
          >
            <div
              class="colOne-header-element-0 col"
            >
              <div>
                <div>
                  One
                </div>
              </div>
            </div>
            <div
              class="colOne-header-element-1 col"
            >
              <div>
                <div>
                  Two
                </div>
              </div>
            </div>
            <div
              class="colOne-header-element-2 col"
            >
              <div>
                <div>
                  Three
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-two d-none col"
        >
          <div
            class="colTwo-direction-wrapper  row"
          >
            <div
              class="colTwo-header-element-0 col"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
